include/tester/ssp/Softphone.hpp

```diff
@@ -1,15 +1,72 @@
 #pragma once
 
+#include<iostream>
+#include<string>
+#include<functional>
+#include<memory>
+
 #include"SSPAccount.hpp"
 #include"SSPCall.hpp"
+#include"SoftphoneArguments.hpp"
 
 class Softphone
 {
 public:
-    Softphone(/* args */);
-    ~Softphone();
+    Softphone(const SoftphoneArguments & args):
+        _account(args.id, args.domain, args.secret, std::bind(&Softphone::onIncomingCall, this, std::placeholders::_1))
+    {
+        _call = std::make_shared<SSPCall>(&_account, std::bind(&Softphone::onCallState, this,
+            std::placeholders::_1, std::placeholders::_2));
+        _account.apply();
+    }
 
-private:
-    std::vector<SSPAccount *> _accounts;
+    ~Softphone() = default;
+
+    void call(const std::string & uri)
+    {
+        _call->callTo(uri);
+    }
+
+    void onCallState(pj::CallInfo ci, const pj::OnCallStateParam &prm)
+    {
+        PJ_UNUSED_ARG(prm);
+        std::cout << "*** Call: " <<  ci.remoteUri << " [" << ci.stateText
+        << "]" << std::endl;
+        if (ci.state == PJSIP_INV_STATE_DISCONNECTED) {
+            /* Delete the call */
+            _call.reset();
+        }
+    }
 
-};
+    void onIncomingCall(const pj::OnIncomingCallParam &iprm)
+    {
+        SSPCall *in_call = new SSPCall(&_account, std::bind(&Softphone::onCallState,
+            this, std::placeholders::_1, std::placeholders::_2), iprm.callId);
+        if(!_call->isActive()){
+            _call = std::make_shared<SSPCall>(std::move(in_call));
```

Comment:
This is an error - there is no signature for an `SSPCall` constructor that takes an `SSPCall *` parameter. therefore - just change the type of `in_call` to a `std::shared_ptr` (don't forget `std::make_shared`), and just assign it to _call (preferably by using the move assignment operator).

Reply:
done